.github/workflows/nightly-test.yaml

```diff
@@ -36,17 +37,25 @@ jobs:
         run: |
           cd build
           snap download k8s --channel=${{ matrix.release }} --basename k8s
+          #create env variables for artifacts
+          echo "os_dash=${{ matrix.os }}" | sed 's/:/-/g' >> $GITHUB_ENV
+          echo "release_dash=${{  matrix.release }}" | sed 's#/#-#g'>> $GITHUB_ENV
```

Comment:
This does not seem like the right place, the env variables are a little messy in their generation and far removed from where they are used and do not fit into the step that installs the snap.